pkg/console/controllers/managedclusters/controller.go

```diff
@@ -0,0 +1,450 @@
+	managedClusterList := []string{}
+	managedClusterListErrors := []string{}
+	for _, managedClusterOAuthView := range managedClusterOAuthClientViews {
+		status, statusErr := managedclusterviewsub.GetStatus(managedClusterOAuthView)
+		if !status || statusErr != nil {
+			namespace, namespaceErr := managedclusterviewsub.GetNamespace(managedClusterOAuthView)
+			if namespaceErr != nil || namespace == "" {
+				managedClusterListErrors = append(managedClusterListErrors, fmt.Sprintf("Unable to create oauth client for cluster %v: managed cluster view status unknown", namespace))
+			} else {
+				managedClusterList = append(managedClusterList, namespace)
+			}
+		}
+	}
+
+	secret, secErr := c.secretsClient.Secrets(api.TargetNamespace).Get(ctx, secretsub.Stub().Name, metav1.GetOptions{})
+	if secErr != nil {
+		return nil, fmt.Errorf("Failed to get secret: %v", secErr), "ManagedClusterActionCreateOAuthClientSyncError"
+	}
+
+	oauthClient, oAuthErr := c.oauthClient.OAuthClients().Get(ctx, oauthsub.Stub().Name, metav1.GetOptions{})
+	if oAuthErr != nil {
+		return nil, fmt.Errorf("Failed to get oauth client: %v", oAuthErr), "ManagedClusterActionCreateOAuthClientSyncError"
+	}
+
+	errs := []string{}
+	managedClusterActionCreateOAuthClients := []*unstructured.Unstructured{}
+	secretString := secretsub.GetSecretString(secret)
+	redirects := oauthsub.GetRedirectURIs(oauthClient)
+	for _, managedClusterName := range managedClusterList {
+		mca := managedclusteractionsub.DefaultCreateOAuthClient(operatorConfig, managedClusterName, secretString, redirects)
+		gvr := managedclusteractionsub.GetGVR()
+		opt := metav1.CreateOptions{}
+		oAuthCreateResp, oAuthCreateErr := c.dynamicClient.Resource(gvr).Namespace(managedClusterName).Create(ctx, mca, opt)
+		if oAuthCreateErr != nil && apierrors.IsAlreadyExists(oAuthCreateErr) {
+			mcaOAuthName, _ := managedclusteractionsub.GetName(mca)
+			oAuthCreateResp, oAuthCreateErr = c.dynamicClient.Resource(gvr).Namespace(managedClusterName).Get(ctx, mcaOAuthName, metav1.GetOptions{})
+		}
+
+		if oAuthCreateErr != nil {
+			errs = append(errs, fmt.Sprintf("Error syncing managed cluster action for oauth client for cluster %s: %v", managedClusterName, oAuthCreateErr))
+		} else {
+			managedClusterActionCreateOAuthClients = append(managedClusterActionCreateOAuthClients, oAuthCreateResp)
+		}
+	}
+
+	if len(errs) > 0 {
+		return nil, errors.New(strings.Join(errs, "\n")), "ManagedClusterActionCreateOAuthClientSyncError"
+	}
+
+	if len(managedClusterListErrors) > 0 {
+		return nil, errors.New(strings.Join(managedClusterListErrors, "\n")), "ManagedClusterActionCreateOAuthClientSyncError"
+	}
+
+	return managedClusterActionCreateOAuthClients, nil, ""
+}
+
+func (c *ManagedClusterController) SyncManagedClusterViewIngressCert(ctx context.Context, operatorConfig *operatorv1.Console) ([]*unstructured.Unstructured, error, string) {
+	managedClusters, listErr := c.listManagedClusters(ctx)
```

Comment:
We list managed clusters at least 3 times per sync loop. We could probably just do this once at the beginning of the sync loop.